src/Logary/Scheduling.fs

```diff
@@ -4,74 +4,90 @@ module Logary.Internals.Scheduling
 
 // creds to Dave Thomas for his F# snippet
 open System.Threading
-  
-open FSharp.Actor
+
+open Hopac
+open Hopac.Infixes
 
 open NodaTime
 
+type Cancellation = private { cancelled: IVar<unit> }
+
+[<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
+module Cancellation =
+  let create () =
+    { cancelled = IVar.Now.create () }
+
+  // https://github.com/logary/logary/pull/92#issuecomment-164139379
+  let isCancelled cancellation = job {
+    return! (cancellation.cancelled ^->. true) <|> (Alt.always false)
+  }
+
+  let cancel cancellation = job {
+    do! IVar.fill cancellation.cancelled ()
```

Comment:
For the particular case of cancellation I would recommend using `IVar.tryFill`.

Note also the comment above about using `Promise<unit>` rather than `IVar<unit>` as the type.  You could then use a design similar to what .Net does by having a cancellation token "source".  The source knows the underlying `IVar`, while the the token itself only exposes it as a `Promise`.

See also [here](https://github.com/Hopac/Hopac/issues/33#issuecomment-67754873).